crates/rpc-proto/proto/responses.proto

```diff
@@ -62,6 +62,20 @@ message SyncStateResponse {
     repeated NullifierUpdate nullifiers = 8;
 }
 
+message SyncNoteResponse {
+    // Number of the latest block in the chain
+    fixed32 chain_tip = 1;
+
+    // Block header of the block with the first note matching the specified criteria
+    block_header.BlockHeader block_header = 2;
+
+    // Proof for block header's MMR with respect to the chain tip.
+    merkle.MerklePath mmr_path = 3;
+
+    // List of all notes together with the Merkle paths from `response.block_header.note_root`
+    repeated note.NoteSyncRecord notes = 4;
+}
+
```

Comment:
Is there a rationale for specifying `optional` in some other responses? e.g. in `GetBlockHeaderByNumberResponse` some of these are optional.

And these would be none if there are no matching notes iiuc.

Reply:
I think we rarely have response fields as optional. In the specific case of `GetBlockHeaderByNumberResponse`, these are optional because we have an optional `include_mmr_proof` filed in the request. If this is set to `false` then, MMR proof-related fields are omitted from the response.